app/controllers/wpcc/your_results_controller.rb

```diff
@@ -53,5 +56,13 @@ def salary_message
         salary_frequency: session[:salary_frequency]
       )
     end
+
+    def default_salary_frequency
+      Hash[
+        Wpcc::YourDetailsForm::SALARY_FREQUENCIES.map do |elem|
+          [elem, elem == 'year' ? 'month' : elem]
+        end
+      ]
+    end
```

Comment:
This two methods about salary frequency, does not appear to be a controller responsibility. Since it is a business rule, can we move to your class? This class will receive the params and the session and base your decision on those. Wdyt?